### DISCUSSION TITLE
Issues with the Favorite button

### DISCUSSION THREAD

**Discussion by vishnukvmd** [ente-io/ente#1164] [Enhancements]
[Moved from Discord]

> When trying to save a photo in favorites by pressing the heart, it is not very fast and sometimes it fails making you have to press it more times
> just see how to make adding to favorites detect more outside the heart and be faster because you have to wait for the heart to appear to be sure that it was saved in favorites, and that is like up to 3 seconds sometimes

It seems like we are now waiting for the network call to succeed, we could short-circuit that and give a better offline-first experience.

**Comment by soenketk** [comment]
Currently there also is no favorite button until the photo has been backed up.
This is a problem when not connected to wifi and backup over mobile data is disabled.

**Comment by fswos** [comment]
Same here
It only works online. Why?
Then when you tap a bit next to the heart the photo changes and the next one gets marked. 
heart must work offline and sync later.

**Comment by Leonardoes22** [comment]
Requiring an internet connection to favorite photos is frustrating, especially while traveling. During trips, it's common to take lots of pictures, but internet access may be limited or unavailable for uploading them. Users should be able to favorite photos locally, with those changes syncing automatically once an internet connection is available. This would provide a much better user experience.
